src/tools/tidy/src/deps.rs

```diff
@@ -246,6 +245,12 @@ impl<'a> From<CrateVersion<'a>> for Crate<'a> {
     }
 }
 
+/// Checks if a given path ends with the pattern /vendor/[exception] where [exception]
+/// is the name of a vendored crate for Rust.
+fn path_contains_exception(path: &Path, parent: &Path, exception: &str) -> bool {
+    return path.ends_with(exception) && parent.ends_with("vendor")
```

Comment:
Either use `return xxx;` (with a return statement and a semicolon) or use `xxx` (a final value expression).  Don't mix them.

In this case, the idiomatic version would be:

`path.ends_with(exception) && parent.ends_with("vendor")`

(no return keyword, no semicolon)